index.js

```diff
@@ -234,21 +234,24 @@
 
     function addTorrent(magnet, dlPath, callback) {
         console.log("Adding: " + magnet);
+        var config = {
+            file_priorities: [],
+            add_paused: false,
+            compact_allocation: true,
+            download_location: dlPath,
+            max_connections: -1,
+            max_download_speed: -1,
+            max_upload_slots: -1,
+            max_upload_speed: -1,
+            prioritize_first_last_pieces: false
+        }
+        var isObj = function (obj) {return obj !== undefined && obj !== null && obj.constructor == Object; }
+        config = (isObj(dlPath) ? dlPath : config);
```

Comment:
I think this code is a little confusing in its intention, although after looking at it I can see that it is to keep things backwards compatible. We could also add a new parameter to the method to take in explicit options, then use those to override the base config on a property by property basis. 

```
var BASE_CONFIG = {
            file_priorities: [],
            add_paused: false,
            compact_allocation: true,
            download_location: "",
            max_connections: -1,
            max_download_speed: -1,
            max_upload_slots: -1,
            max_upload_speed: -1,
            prioritize_first_last_pieces: false
}

function addTorrent(magnet, dlPath, callback, opts) {
  var config = Object.assign({}, { download_location: dlPath }, BASE_CONFIG)
  if (opts) {
    config = Object.assign({}, config, opts)
  }
}
```

Reply:
Adding a parameter doesn't make sense because we're then forcing the user to specify the dlPath AND opts if they want to add their own config.

The code made with backwards compatible in mind, although it's probably better to do the Object.assign(), so that you can only insert the options you want to change rather than the whole thing.